src/middlewares/authentication.js

```diff
@@ -0,0 +1,32 @@
+import { verifyToken } from '../helpers/JWT';
+import responseHandler from '../helpers/responseHandler';
+
+/**
+   * Method to generate token
+   * @param {object} req
+   * @param {object} res
+   * @param {Function} next
+   * @returns {Function} next middleware
+   */
+export default function (req, res, next) {
+  const token = req.headers.authorization;
+  if (!token) {
+    return responseHandler.error(res, 'Please signin to continue', 401);
+  }
+  try {
+    const { key } = verifyToken(token);
```

Comment:
Good work bro. Don't you consider refactoring your verifyToken as after  part of middleware instead of flooding your scope with if-else

Reply:
The verifyToken function returns a promise, so error handling occurs at the point where called. Also, in the catch block, I'm checking different situations in the catch block so as to return a proper message, I could abstract it out, but I'm open to other ideas.

Reply:
abstract it as a function  and handle the promise and the error gracefully. passes the decoded token into the req and call the next function.

Reply:
I will look into this.